frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts

```diff
@@ -50,7 +48,7 @@ export const useK8sWatchResources: UseK8sWatchResources = (initResources) => {
   const prevK8sModels = usePrevious(allK8sModels);
   const prevResources = usePrevious(resources);
 
-  const k8sModelsRef = React.useRef<ImmutableMap<string, K8sModel>>(ImmutableMap());
+  const k8sModelsRef = React.useRef<Iterable<string, K8sModel>>(Iterable(allK8sModels));
```

Comment:
Because here https://github.com/openshift/console/blob/6db2de0cab7076af3557a4d6397b43d15ccded15/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts#L66 the `filter` returns an `Iterable` and not a `Map`.

Reply:
But previously it was initialized with an empty map and now its initialized with a copy or allK8sModels? I need to take a deeper look as well.

Reply:
I'm fine with `Iterable` as a type, but can we keep `ImmutableMap()` on the right side so that this PR doesn't change the implementation?

```suggestion
  const k8sModelsRef = React.useRef<Iterable<string, K8sModel>>(ImmutableMap());
```

Reply:
Not sure why it was a Map previously as it was showing errors to me. But instead of allK8sModels it should be an empty map here. I will fix that.
```suggestion
  const k8sModelsRef = React.useRef<Iterable<string, K8sModel>>(Iterable(ImmutableMap()));
```